### PR TITLE
Fix testLogsNotFlushedImmediatelyWhenIntervalIsCustom

### PR DESCRIPTION
**Pull request by russelarms**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

testLogsNotFlushedImmediatelyWhenIntervalIsCustom is continuously failing in merge build but couldn't reproduce on local machine. Please investigate the failure and fix it.

## Related PRs or issues

[AB#62895](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/62895)

### CODE REVIEW

File: AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m [hunk 1]
```diff
@@ -277,38 +277,35 @@ - (void)testLogsFlushedImmediatelyWhenIntervalIsOver {
 }
 
 - (void)testLogsNotFlushedImmediatelyWhenIntervalIsCustom {
-
+
```
Comment (guperrot) [inline AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m]: This PR is adding whitespaces, please run the clang formatter and remove them

Comment (Jamminroot) [reply AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m]: Well, I there is a comment on a next line; As for whitespaces - that Is what I get from clang-formatter

File: AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m [hunk 2]
```diff
@@ -283,19 +283,19 @@ - (void)testLogsNotFlushedImmediatelyWhenIntervalIsCustom {
...
                                                                  flushInterval:flushInterval
                                                                 batchSizeLimit:batchSizeLimit
                                                            pendingBatchesLimit:3];
 
   // When
-  for (NSUInteger i = 0; i < itemsToAdd; i++) {
-    [self.sut enqueueItem:[self getValidMockLog] flags:MSFlagsDefault];
-  }
-  [self enqueueChannelEndJobExpectation];
+  dispatch_async(self.logsDispatchQueue, ^{
```
Comment (MatkovIvan) [inline AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m]: I don't see how it can help. Could you please explain the idea?

Comment (Jamminroot) [reply AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m]: Has to be investigated yet, why this actually fixes. Basically I wanted to check if its some kind of race condition (can't explain where exactly still)

**Comment by MatkovIvan** [comment]
OK, the pass rate 28/30. I'm closing it since we don't have a reliable fix now.

**Pull request closed without merging**